### PR TITLE
New AutoGluon Webpage

### PR DESCRIPTION
**Pull request by gidler**
*Description of changes:*

This change is a major facelift on the [AutoGluon webpage](https://auto.gluon.ai/new/). It allows AutoGluon to leave behind the outdated version of [d2l-book](https://github.com/zhanghang1989/d2l-book) that the webpage was tied to, along with limitations on the version of Sphinx that could be used. We now use the [sphinx furo](https://sphinx-design.readthedocs.io/en/furo-theme/) theme, but can be more easily changed in the future by choosing other [sphinx-design](https://sphinx-design.readthedocs.io) themes.

This change touches most files in the `docs` directory. I intend to split the review of this PR across:
- changes in `.github` directory
- changes to website config and build process
- changes in `tutorials` directory which can be inspected by different vertical teams

Tutorials are now maintained in `ipynb` notebooks rather than `md` files. This change come with significant pros and cons:
*Pros*:
- easy for users to download and use AutoGluon tutorials directly from repository
- easy to maintain notebooks that can be opened in Colab or Studio lab

*Cons*:
- authoring and reviewing `ipynb` files is more difficult (using [this tool](https://www.reviewnb.com/) can help mitigate this)

When reviewing these changes, it is probably easier to inspect tutorials on [the new webpage](https://auto.gluon.ai/new/) than it will be to look through raw files and diffs.

This PR is to merge the new website onto master/dev. Once we are comfortable with it, we will merge onto stable as well.

This checklist of items needs to be done before merge:
- [x] AutoGluon 0.7 Release
- [x] Remove `new` branch url from Colab/Studio Lab links 
- [x] Update 0.6.2 to 0.7.0 in new docs
- [x] Add/remove `pre-` from correct locations in Install page

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/tabular/tabular-quick-start.ipynb [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    "Once we have a predictor that is fit on the training dataset, we can load a separate set of data to use for prediction and evaulation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2022-07-09T05:32:02.338844Z",
+     "start_time": "2022-07-09T05:32:01.721431Z"
```
[the remark this answers is not in the document]
Comment (Innixma) [reply docs/tutorials/tabular/tabular-quick-start.ipynb:217]: You can refer to the `tabular-indepth.ipynb` file source to see what a clean notebook looks like

[the remark this answers is not in the document]
Comment (Innixma) [reply docs/tutorials/tabular/tabular-quick-start.ipynb:217]: probably easiest to re-create the file and just copy the cells into the new file.

File: docs/tutorials/tabular/tabular-quick-start.ipynb [hunk 2]
```diff
@@ -0,0 +1,345 @@
+{
```
Comment (Innixma) [inline docs/tutorials/tabular/tabular-quick-start.ipynb:1]: This quick start is pretty good. We may want to consider using titanic instead of this example (although we definitely should keep this example for at least 1 tutorial, since it is quite good). For the purposes of an initial merge, I am ok with this tutorial.

The main thing we should add to this tutorial is a comparison of how simple this solution is + how strong the result is compared to DeepMind's original solution that was on the cover of nature (300+ lines of code & 80% accuracy with DeepMind vs 3 lines of code & 95% accuracy with AutoGluon). This comparison shouldn't be present in a quick-start tutorial, which is why we should wait to add that comparison until we make a different quick-start tutorial for Tabular.

File: docs/api/autogluon.tabular.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+
+.. currentmodule:: autogluon.tabular
+
+.. autosummary::
+    :nosignatures:
+    :toctree: _autogen
+
+    TabularDataset
+    TabularPredictor
```
Comment (Innixma) [inline docs/api/autogluon.tabular.rst:12]: This is awesome. However, the URL is very strange:

http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-2924/a5fb2a6/api/_autogen/autogluon.tabular.TabularPredictor.html#autogluon.tabular.TabularPredictor.clone

Compared to install URL:

http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-2924/a5fb2a6/install.html

Lets just focus on the suffix:

api/_autogen/autogluon.tabular.TabularPredictor.html#autogluon.tabular.TabularPredictor.clone

install.html

## Oddity 1

`api/_autogen` seems a bit odd.

Here is the URL for pandas `Series.to_frame`:

https://pandas.pydata.org/docs/reference/api/pandas.Series.to_frame.html

They don't have any `_autogen` in the path

## Oddity 2

`autogluon.tabular.TabularPredictor.html#autogluon.tabular.TabularPredictor.clone`

This is quite verbose.

Why not:

`autogluon.tabular.TabularPredictor.html#clone`

Alternatively, can we get a separate page for every method like is done with Pandas?

https://pandas.pydata.org/docs/reference/api/pandas.Series.to_frame.html

This looks really nice and is super easy to navigate.

The URL for a standalone page could be `auto.gluon.ai/stable/api/autogluon.tabular.TabularPredictor.clone.html`

**Review by Innixma — COMMENTED** [review]
Awesome work! Added comments.

In general, Tabular LGTM!

File: docs/install.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+```bash
+conda create -n autogluon python=3.9 -y
+conda activate autogluon
+```
+::::
+
+[pip](https://pip.pypa.io/en/stable/installation/) is the primary way to install AutoGluon, and a recent version is required by AutoGluon's dependencies:
+
+```bash
+python -m pip install --upgrade pip
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/install.md]: I've seen a couple python packages suggest `python -m pip` for the reasons discussed [here](https://stackoverflow.com/questions/60782785/python3-m-pip-install-vs-pip3-install). I did a quick survey of packages in a similar area to us (pandas, numpy, scikit) and they all use simple `pip install`, so I'm fine keeping it simple here and we can change later if we get feedback otherwise.

[the remark this answers is not in the document]
Comment (Innixma) [reply docs/install.md]: Sounds good, lets go with `pip install` initially as you mentioned. I believe I was the one that switched to using `python -m` at one point early on because of a user getting confused, but it has been several years and I think the likelihood it will cause issues is a lot less than back then.

File: .github/workflow_scripts/build_all_docs.sh [hunk 5]
```diff
@@ -53,28 +53,25 @@ else
...
 
-mkdir -p docs/_build/rst/tutorials/
-aws s3 cp $BUILD_DOCS_PATH docs/_build/rst/tutorials/ --recursive
-
 setup_build_contrib_env
 install_all_no_tests
 
-sed -i -e "s@###_PLACEHOLDER_WEB_CONTENT_ROOT_###@http://$site@g" docs/config.ini
-sed -i -e "s@###_OTHER_VERSIONS_DOCUMENTATION_LABEL_###@$other_doc_version_text@g" docs/config.ini
-sed -i -e "s@###_OTHER_VERSIONS_DOCUMENTATION_BRANCH_###@$other_doc_version_branch@g" docs/config.ini
```
Comment (yinweisu) [inline .github/workflow_scripts/build_all_docs.sh:64]: With the new website how do we generate link to stable/dev one dynamically?

**Review by yinweisu — APPROVED** [review]
LGTM in general regarding the workflow change. Had one minor comment

File: .github/workflow_scripts/build_all_docs.sh (hunk 5 above)
Comment (gidler) [reply .github/workflow_scripts/build_all_docs.sh:64]: These ad-hoc replacement strings were used in the nav-bar of the old website to link to things like versions, api, tutorials, etc. These exact links don't exist on the new webpage, but their equivalent sections are handled by the sphinx build process automatically. I believe these only needed to be done this way because the links were hard coded in the config file, not built by sphinx.

**Review by tonyhoo — COMMENTED** [review]
LGTM in general. Thanks for the change and the new website looks great! Some minor comments regarding installation guide and build process

File: docs/install-gpu-pip.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+```console
+pip install -U pip
+pip install -U setuptools wheel
+
+# Install the proper version of PyTorch following https://pytorch.org/get-started/locally/
+pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116
```
Comment (Innixma) [inline docs/install-gpu-pip.md:6]: I thought we didn't need to manually install torch for GPU? Shouldn't it install automatically via `pip install autogluon`?

File: docs/install-gpu-source.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+```console
+pip install -U pip
+pip install -U setuptools wheel
+
+# Install the proper version of PyTorch following https://pytorch.org/get-started/locally/
+pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116
```
Comment (Innixma) [inline docs/install-gpu-source.md:6]: I thought we didn't need to manually install torch for GPU? Shouldn't it install automatically?

File: docs/install.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+
+```bash
+pip install --pre autogluon
+```
+:::
+
+
+:::{dropdown} M1 and M2 Apple Silicon
+
+Apple Silicon is now supported via the `conda` installation instructions outlined above. `conda-forge` will install the GPU version if a user's machine supports it.
```
Comment (Innixma) [inline docs/install.md:227]: @gradientsky please confirm if this is correct

**Review by Innixma — APPROVED** [review]
In general LGTM! We can make smaller PRs for additional improvements & follow-ups after this is merged. Amazing work!

File: docs/install-gpu-pip.md (hunk 6 above)
Comment (gidler) [reply docs/install-gpu-pip.md:6]: My intention was for these to be the exact same as they are on the existing website. I am not sure if your statement about not needing to manually install torch for GPU is correct but can look into it later.

File: docs/cheatsheet.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+## Multimodal
+
+```{image} https://automl-mm-bench.s3-accelerate.amazonaws.com/cheatsheet/v0.7.0/AutoGluon_Multimodal_Cheatsheet_v0.7.0.png
+:width: 900
+```
+
+Download the PDF version with clickable links [multimodal-cheatsheet].
+
+## Time Series
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/cheatsheet.md:21]: An EDA Cheat Sheet did not exist last time I synced with Alex. I'm not sure if Alex is planning on making one in the future.

File: docs/install-mac-libomp.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+:::{note} LightGBM support on MacOS (LibOMP)
+AutoGluon dependency LightGBM uses `libomp` for multi-threading. If you install `libomp` via `brew install libomp`, you may get segmentation faults due to incompatible library versions. Install a compatible version using the following commands:
+
+```bash
+# Uninstall libomp if it was previous installed
+brew uninstall -f libomp
+wget https://raw.githubusercontent.com/Homebrew/homebrew-core/fb8323f2b170bd4ae97e1bac9bf3e2983af3fdb0/Formula/libomp.rb
+brew install libomp.rb
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/install-mac-libomp.md:8]: Good catch, I will add in next push

File: .github/workflow_scripts/build_all_docs.sh [hunk 11]
```diff
@@ -53,28 +53,25 @@ else
...
+LOCAL_DOC_PATH=_build/html
+
+cd docs
+sphinx-build -D nb_execution_mode=off -b html . $LOCAL_DOC_PATH
```
[the remark this answers is not in the document]
Comment (gidler) [reply .github/workflow_scripts/build_all_docs.sh:62]: Yes, the previous `build-*-tutorial` CI steps generate the tutorials with `nb_execution_mode=force`. Then in the next line of this script, we copy those tutorials with the execution results over these.

File: docs/ReleaseInstructions.md [hunk 12]
```diff
@@ -1,8 +1,8 @@
 # Release process
 
-### Prior to release: 1 week out
+## Prior to release: 1 week out
 
-* Ensure the version specified in `docs/config.ini`, `VERSION`, `docs/index.rst`, and `docs/badges.rst` align with the intended release version.
+* Ensure the version specified in `docs/conf.py`, `VERSION`, and `docs/index.md` align with the intended release version.
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/ReleaseInstructions.md:5]: No, we don't have/use the `docs/badges` file anymore on the new webpage. Additionally, I don't think it was used in the old webpage either, because basically the exact same lines that were in the `docs/badges.rst` file were also in the `docs/index.rst` file, and `docs/badges.rst` isn't imported anywhere in our code.

File: docs/install.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+You can also specify optional dependencies:
+
+```bash
+python -m pip install autogluon.tabular[lightgbm,catboost]
+```
+
+Modules and optional dependencies:
+
+1. `autogluon.tabular`: functionality for tabular data (TabularPredictor)
+   - Optional dependencies included by default: `lightgbm`,`catboost`,`xgboost`,`fastai`.
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/install.md]: I updated the modularized install instructions on my most recent push.

File: docs/install.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+   - Optional dependencies not included by default: `vowpalwabbit`, `skex`. The later will speedup KNN training and inference on CPU by 25x.
```
[the remark this answers is not in the document]
Comment (gidler) [reply docs/install.md]: I updated the modularized install instructions on my most recent push.

**Review by tonyhoo — APPROVED** [review]
LGTM!

File: docs/install-modules.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
+AutoGluon is modularized into [sub-modules](https://packaging.python.org/guides/packaging-namespace-packages/) specialized for tabular, multimodal, or time series data. You can reduce the number of dependencies required by solely installing a specific sub-module via:  `pip install <submodule>`, where `<submodule>` may be one of the following options:
+
+- `autogluon.tabular` - functionality for tabular data (TabularPredictor)
+    - The default installation of `autogluon.tabular` standalone is a skeleton installation.
+    - Install via `pip install autogluon.tabular[all]` to get the same installation of tabular as via `pip install autogluon`
+    - Available optional dependencies: `lightgbm,catboost,xgboost,fastai`. These are included in `all`.
```
Comment (Innixma) [inline docs/install-modules.md:6]: also `ray` is included in `all`

File: docs/install-modules.md [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    - Optional dependencies not included in `all`: `vowpalwabbit,imodels,skex,skl2onnx`.
+    - To run `autogluon.tabular` with only the optional LightGBM and CatBoost models for example, you can do: `pip install autogluon.tabular[lightgbm,catboost]`
+    - Experimental optional dependency: `skex`. This will speedup KNN models by 25x in training and inference on CPU. Use `pip install autogluon.tabular[all,skex]` to enable, or `pip install "scikit-learn-intelex<2021.5"` after a standard installation of AutoGluon.
```
Comment (Innixma) [inline docs/install-modules.md:9]: We can remove the `pip install "scikit-learn-intelex<2021.5"`, easier for them to just specify the autogluon pip install, which also works after a standard installation of AutoGluon.

File: docs/install-modules.md [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    - Optional Dependency: `vowpalwabbit`. This will install the VowpalWabbit package and allow you to fit VowpalWabbit in TabularPredictor.
+    - Optional Dependency: `imodels`. This will install the imodels package and allow you to fit interpretable models in TabularPredictor.
+    - Optional Dependency: `skl2onnx`. This will enable model compilation support via `predictor.compile_models()` on supported models.
```
Comment (Innixma) [inline docs/install-modules.md:12]: `This will enable model compilation support via` -> `This will enable ONNX model compilation via`

File: docs/install-modules.md [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+    - The default installation of `autogluon.tabular` standalone is a skeleton installation.
+    - Install via `pip install autogluon.tabular[all]` to get the same installation of tabular as via `pip install autogluon`
+    - Available optional dependencies: `lightgbm,catboost,xgboost,fastai`. These are included in `all`.
+    - Optional dependencies not included in `all`: `vowpalwabbit,imodels,skex,skl2onnx`.
+    - To run `autogluon.tabular` with only the optional LightGBM and CatBoost models for example, you can do: `pip install autogluon.tabular[lightgbm,catboost]`
+    - Experimental optional dependency: `skex`. This will speedup KNN models by 25x in training and inference on CPU. Use `pip install autogluon.tabular[all,skex]` to enable, or `pip install "scikit-learn-intelex<2021.5"` after a standard installation of AutoGluon.
+    - Optional Dependency: `vowpalwabbit`. This will install the VowpalWabbit package and allow you to fit VowpalWabbit in TabularPredictor.
+    - Optional Dependency: `imodels`. This will install the imodels package and allow you to fit interpretable models in TabularPredictor.
```
Comment (Innixma) [inline docs/install-modules.md:11]: Experimental optional dependency

**Review by Innixma — APPROVED** [review]
Looks great!! Added some minor comments
